#DPATCHLEVEL=0

2005-12-28  Samuel Thibault  <samuel.thibault@ens-lyon.org>

	* i386/i386/iopb.c (i386_io_port_remove): Fix unlocking when io mapping
	is not found.


Index: i386/i386/iopb.c
===================================================================
RCS file: /cvsroot/hurd/gnumach/i386/i386/Attic/iopb.c,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 iopb.c
--- i386/i386/iopb.c	25 Feb 1997 21:27:09 -0000	1.1.1.1
+++ i386/i386/iopb.c	28 Dec 2005 16:18:40 -0000
@@ -468,6 +468,10 @@
 	/*
 	 * No mapping.
 	 */
+
+	simple_unlock(&pcb->lock);
+	simple_unlock(&iopb_lock);
+
 	return KERN_INVALID_ARGUMENT;
 }
 
